src/hotspot/share/gc/g1/g1Arguments.cpp

```diff
@@ -177,6 +177,25 @@ void G1Arguments::initialize() {
     FLAG_SET_ERGO(ParallelGCThreads, 1);
   }
 
+  if (G1BarrierSimple) {
+#if !defined(_LP64) || !(defined(X86) || defined(AARCH64))
```

Comment:
X86 is allowed here?

Reply:
only x86 32bit is not supported.

Reply:
ok, took me a while to see this means !(x86_64 || aarch64)